src/Core/Query/Helper.php

```diff
@@ -462,8 +461,24 @@ public function filterControlCharacters(string $data): string
     }
 
     /**
-     * Render placeholders in a querystring.
+     * Escape text for use as parsed character data content in an XML element.
+     *
+     * This escapes characters that can't appear as character data using their
+     * corresponding entity references. Per the definition of XML, "&" and "<"
+     * MUST be escaped when used in character data, ">" MUST be escaped in the
+     * string "]]>" and MAY be otherwise, so we escape it to be safe.
      *
+     * @param string $data
+     *
+     * @return string
+     */
+    public function escapePCDATAContent(string $data): string
+    {
+        return str_replace(['&', '<', '>'], ['&amp;', '&lt;', '&gt;'], $data);
```

Comment:
I don't see an advantage here. According to the PHP manual, this function is exactly made for this:
`htmlspecialchars($value, ENT_NOQUOTES)`

Reply:
Actually, according to the PHP manual, this functions is:
`htmlspecialchars($value, ENT_XML1)`
I must admit I only found out just now.

My main problem with `htmlspecialchars` was that the encoding needs to be passed explicitly as a third parameter if it's something other than `ini_get("default_charset")`. It was used in places where I had no way to directly access the input encoding from the request.

Another possible issue is that's it's limited to a number of encodings. Not all standard charset for Java are supported. And I don't see GB18030 among the supported ones, which is used in the `techproducts` testdata.

However this turns out, having it in a helper function means we only have to change it in one place if we decide on a different implementation.

Reply:
Got it. In this case we extend the comments for this function and its purpose.

Reply:
Should I add comments inside the function explaining why it doesn't use `htmlspecialchars`?

The PHPdoc already explains exactly what to expect from the function. From the caller's perspective, that's all that's needed. It's also exactly what the unit test is testing. Implementation details shouldn't matter on this level.

Reply:
We should add one sentence, why htmlspecialchars isn't used because not all character sets are supported.

Reply:
I've added the comment.